### DISCUSSION TITLE
Additional component logic

### DISCUSSION THREAD

**Discussion by d-henn** [gregberge/twc#44] [Q&A] [answered]
Is it possible to override the logic to a component? For example, a `Button` component that I need to perform something additional in the `onClick`?

Basic example with a standard non `twc` component may look like:

```js

function Button(props: ButtonProps) {
  const { onClick } = props

  function customClick(event: React.MouseEventHandler<HTMLButtonElement> | undefined) {
    // do something on any button click
    console.log("Good job!")
    
    // then perform original onClick if present
    if (onClick) {
      onClick(event)
    }
  }

  return <button {...props} onClick={customClick} />
}
```

**Comment by gregberge** [comment, marked as answer]
No it's not the goal of this library if you need to do that then you should use a functional component.
